Libraries/LibWeb/TrustedTypes/TrustedTypePolicy.cpp

```diff
@@ -169,4 +186,158 @@ WebIDL::ExceptionOr<JS::Value> TrustedTypePolicy::get_trusted_type_policy_value(
+    //    arguments:
+    //      « trustedTypeName, sink »
+    //    throwIfMissing:
+    //      false
+    //  3. If the algorithm threw an error, rethrow the error and abort the following steps.
+    auto arguments = GC::RootVector<JS::Value>(vm.heap());
+    arguments.append(JS::PrimitiveString::create(vm, to_string(trusted_type_name)));
+    arguments.append(JS::PrimitiveString::create(vm, to_string(sink)));
+    auto policy_value = TRY(default_policy->get_trusted_type_policy_value(
+        trusted_type_name,
+        input.visit(
+            [](auto& value) { return value->to_string(); },
+            [](Utf16String& value) { return value; }),
+        arguments,
+        ThrowIfCallbackMissing::No));
+
+    //  4. If policyValue is null or undefined, return policyValue.
+    if (policy_value.is_nullish())
+        return Optional<TrustedType> {};
+
+    //  5. Let dataString be the result of stringifying policyValue.
+    Utf16String data_string;
+    switch (trusted_type_name) {
+    case TrustedTypeName::TrustedHTML:
+    case TrustedTypeName::TrustedScript:
+        data_string = TRY(WebIDL::to_utf16_string(vm, policy_value));
+        break;
+    case TrustedTypeName::TrustedScriptURL:
+        data_string = TRY(WebIDL::to_utf16_usv_string(vm, policy_value));
+        break;
+    default:
+        VERIFY_NOT_REACHED();
+    }
+
+    //  6. Return a new instance of an interface with a type name trustedTypeName, with its associated data value set to dataString.
+    switch (trusted_type_name) {
+    case TrustedTypeName::TrustedHTML:
+        return realm.create<TrustedHTML>(realm, move(data_string));
+    case TrustedTypeName::TrustedScript:
+        return realm.create<TrustedScript>(realm, move(data_string));
+    case TrustedTypeName::TrustedScriptURL:
+        return realm.create<TrustedScriptURL>(realm, move(data_string));
+    }
+    VERIFY_NOT_REACHED();
+}
+
+// https://www.w3.org/TR/trusted-types/#get-trusted-type-compliant-string-algorithm
+WebIDL::ExceptionOr<Utf16String> get_trusted_type_compliant_string(TrustedTypeName expected_type, JS::Object& global, Variant<GC::Root<TrustedHTML>, GC::Root<TrustedScript>, GC::Root<TrustedScriptURL>, Utf16String> input, InjectionSink sink, String sink_group)
+{
+    // 1. If input is an instance of expectedType, return stringified input and abort these steps.
+    switch (expected_type) {
+    case TrustedTypeName::TrustedHTML:
+        if (auto* const value = input.get_pointer<GC::Root<TrustedHTML>>(); value)
+            return (*value)->to_string();
+        break;
+    case TrustedTypeName::TrustedScript:
+        if (auto* const value = input.get_pointer<GC::Root<TrustedScript>>(); value)
+            return (*value)->to_string();
+        break;
+    case TrustedTypeName::TrustedScriptURL:
+        if (auto* const value = input.get_pointer<GC::Root<TrustedScriptURL>>(); value)
+            return (*value)->to_string();
+        break;
+    }
```

Comment:
nit: We typically put the `*` after `const`:
```suggestion
    switch (expected_type) {
    case TrustedTypeName::TrustedHTML:
        if (auto const* value = input.get_pointer<GC::Root<TrustedHTML>>(); value)
            return (*value)->to_string();
        break;
    case TrustedTypeName::TrustedScript:
        if (auto const* value = input.get_pointer<GC::Root<TrustedScript>>(); value)
            return (*value)->to_string();
        break;
    case TrustedTypeName::TrustedScriptURL:
        if (auto const* value = input.get_pointer<GC::Root<TrustedScriptURL>>(); value)
            return (*value)->to_string();
        break;
    }
```